applications/ConvectionDiffusionApplication/custom_utilities/bfecc_convection.h

```diff
@@ -309,14 +395,49 @@ template<std::size_t TDim> class BFECCConvection
             {
                 for(unsigned int ii=node_partition[kkk]; ii<node_partition[kkk+1]; ii++)
                 {
-                    ModelPart::NodesContainerType::iterator inode = inodebegin+ii;
-                    inode->GetSolutionStepValue(rVar,1) = inode->FastGetSolutionStepValue(rVar);
+                        ModelPart::NodesContainerType::iterator inode = inodebegin+ii;
+
+                        if (inode->IsFixed(rVar))
+                        {
+                            inode->FastGetSolutionStepValue(rVar)=inode->GetSolutionStepValue(rVar,1);
+                        }
                 }
             }
+
             KRATOS_CATCH("")
+    }
+
+    void CopyScalarVarToPreviousTimeStep(ModelPart& rModelPart, const Variable< double >& rVar)
+    {
+        KRATOS_TRY
+        ModelPart::NodesContainerType::iterator inodebegin = rModelPart.NodesBegin();
```

Comment:
Same as above, since this part of the code was already implemented and not modified here, maybe, it is better to create a separate PR for its modification.